src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -80,61 +171,61 @@ public void postProcess(File file) {
 
     public final void updateMetadata(Path outputFile) {
         try {
-            NetcdfFileWriter nc = NetcdfFileWriter.openExisting(outputFile.toAbsolutePath().toString());
+            String location = outputFile.toAbsolutePath().toString();
+            NetcdfFileWriter nc = NetcdfFileWriter.openExisting(location);
+            GridDataset gridDs = GridDataset.open(location);
 
             nc.setRedefineMode(true);
             for (Attribute newAttr : getGlobalAttributesToUpdate(nc.getNetcdfFile())) {
                 nc.addGroupAttribute(null, newAttr);
             }
             nc.setRedefineMode(false);
             nc.close();
-        }
-        catch (IOException e) {
+        } catch (IOException e) {
             throw new GoGoDuckException(String.format("Failed updating metadata for file '%s': '%s'", outputFile, e.getMessage()));
         }
     }
 
-    public NcksSubsetParameters getNcksSubsetParameters() {
-        NcksSubsetParameters ncksSubsetParameters = new NcksSubsetParameters();
-        ncksSubsetParameters.put("LATITUDE", subset.get("LATITUDE"));
-        ncksSubsetParameters.put("LONGITUDE", subset.get("LONGITUDE"));
-        ncksSubsetParameters.addTimeSubset("TIME", subset.get("TIME"));
-        return ncksSubsetParameters;
-    }
+    public NcksSubsetParameters getNcksSubsetParameters(String location) {
+        CoordinateAxis time = null, latitude = null, longitude = null;
+        GridDataset gridDs = null;
 
-    public static GoGoDuckModule newInstance(String profile, IndexReader indexReader, String subset, UserLog userLog) {
-        String thisPackage = GoGoDuckModule.class.getPackage().getName();
-        String classToInstantiate = String.format("GoGoDuckModule_%s", profile);
-
-        GoGoDuckModule module = null;
-        while (null == module && !classToInstantiate.isEmpty()) {
-            logger.debug(String.format("Trying class '%s.%s'", thisPackage, classToInstantiate));
-            try {
-                Class classz = Class.forName(String.format("%s.%s", thisPackage, classToInstantiate));
-                module = (GoGoDuckModule) classz.newInstance();
-                module.init(profile, indexReader, subset, userLog);
-                logger.info(String.format("Using class '%s.%s'", thisPackage, classToInstantiate));
-                return module;
-            }
-            catch (Exception e) {
-                logger.debug(String.format("Could not find class for '%s.%s'", thisPackage, classToInstantiate));
+        try {
```

Comment:
Perhaps do this once in setup - we've got real problems if variables are named differently in each file.